libsolidity/inlineasm/AsmParser.cpp

```diff
@@ -260,6 +261,17 @@ assembly::Expression Parser::parseExpression()
 				boost::lexical_cast<string>(args) +
 				" arguments)"
 			));
+		// Disallow instructions returning multiple values and DUP/SWAP
+		if (
+			instructionInfo(instr.instruction).ret != 1 ||
+			isDupInstruction(instr.instruction) ||
+			isSwapInstruction(instr.instruction))
```

Comment:
Explicit dup/swap check here is not required as it is covered by `ret != 1`, but it still may be beneficial to have it explicitly for readability.

Reply:
Yep, I agree! I think you have a suplus `)` here, though.